graphics/surface.h

```diff
@@ -86,6 +86,13 @@ struct Surface {
 	 * Pixel data of the surface.
 	 */
 	void *pixels;
+#ifdef ATARI
+	/**
+	 * Unaligned pixel data of the surface.
+	 * Used in Surface::free() in blit_atari.cpp.
+	 */
+	void *pixelsUnaligned;
```

Comment:
It's possible to avoid this additional variable by storing the unaligned address besides the data.
The following snippet (inspired from backends/plugins/riscos/riscos-provider.cpp) does this:
```
		void *p = malloc(size + sizeof(uintptr) + align - 1);
		void *np = (void *)(((uintptr)p + sizeof(uintptr) + align - 1) & (-align));

		*((uintptr *)p-1) = (uintptr)p;
```
The free is done like this:
```
		free((void *)*((uintptr *)ptr - 1));
```

Reply:
Oh, nice! I have to say, this is pretty inventive! I don't see a reason why not to take this path, thanks.